Core/clim-basic/windowing/ports.lisp

```diff
@@ -256,8 +256,8 @@ is a McCLIM extension.")
              (new-event (shallow-copy-object event event-class)))
         (when (typep new-event 'pointer-event)
           (get-pointer-position (target-sheet new-event)
-            (setf (slot-value new-event 'x) x
-                  (slot-value new-event 'y) y)))
+            (setf (slot-value new-event 'sheet-x) x
+                  (slot-value new-event 'sheet-y) y)))
```

Comment:
(I wanted to attach this to the correct commit, but not necessarily to this code location)

I think that several places would have to be adapted to this change:
* One is `throw-object-ptype` which produced an event with unbound `sheet-x` slot when i tried this branch.
* Backend methods on `synthesize-pointer-motion-event`. I'm not sure whether those can work unmodified or not.
* Are there other places that directly construct pointer events?

Alternatively, can sensible default values for the new slots be computed? Maybe at least default values good enough for synthesized events?

Reply:
how could that be? we initialize sheet-x and sheet-y in initialize-instance based on x/y initargs:
```
(defmethod initialize-instance :after ((event device-event) &key x y)
  (when-let ((sheet (event-sheet event)))
    (multiple-value-bind (sheet-x sheet-y)
        (untransform-position (sheet-native-transformation sheet) x y)
      (setf (slot-value event 'sheet-x) (floor sheet-x)
            (slot-value event 'sheet-y) (floor sheet-y)))))
```
could you provide a test case with which I could reproduce the issue?

Reply:
I've made them default to X/Y, so if the sheet is not supplied when making the event instance it will fallback to that:

```
(defmethod initialize-instance :after ((event device-event) &key x y)
  (if-let ((sheet (event-sheet event)))
    (multiple-value-bind (sheet-x sheet-y)
        (untransform-position (sheet-native-transformation sheet) x y)
      (setf (slot-value event 'sheet-x) (floor sheet-x)
            (slot-value event 'sheet-y) (floor sheet-y)))
    (setf (slot-value event 'sheet-x) x
          (slot-value event 'sheet-y) y)))
```

Reply:
Why are the `floor` calls necessary?

Reply:
beats me, fixed.